### DISCUSSION TITLE
Backend Capabilities in laravel context

### DISCUSSION THREAD

**Discussion by simonbuehler** [soketi/pws#324] [Q&A] [answered]
hi,

coming from the  beyondcode/laravel-websockets package i have a few questions  regarding the capabilities of this package in a laravel context

is it possible  to implement logic that

* select a random user from a presence channel
* get back the complete user object 
* send messages from the backend to a client without a client having initalised a conversation ?

any insight would be highly appreciated!

**Comment by rennokki** [comment, marked as answer]
The first two are possible via the [API's `/users` endpoint](https://pusher.com/docs/channels/library_auth_reference/rest-api/#get-users). However, you can only get the presence info that's shared to the server when a user connects to a presence channel.

```php
Broadcast::channel('orders.{order}', function ($user) {
    return ['id' => $user->id]; // or whatever
});

$client = new Pusher(...);

// https://github.com/pusher/pusher-http-php#get-user-information-from-a-presence-channel
$client->getPresenceUsers('presence-channel');
```

The last one can be achieved with a custom implementation in the project's API. Currently, the server cannot be configured, but this seems like a worthy feature to have. I might come up with an implementation and discuss on the PR once i mention it. :D

**Comment by rennokki** [comment]
@simonbuehler What I might think to workaround the last point, and would make the feature redundant: you can create a channel for the specific user.

```php
Broadcast::channel('user.{user}', function ($user, $id) {
    return $id === $user->id;
});
```

Then have an event that broadcasts to that channel. The only thing to implement yourself is the listening to that private channel, so that just the assigned user has access to it.
